sky/engine/core/script/dart_init.cc

```diff
@@ -341,6 +328,74 @@ void* _DartSymbolLookup(const char* symbol_name) {
   return DartLookupSymbolInLibrary(symbol_name, nullptr);
 }
 
+#elif OS(ANDROID)
+
+// Describes an asset file that holds a part of the precompiled snapshot.
+struct SymbolAsset {
+  SymbolAsset(const char* file, bool exec)
+      : file_name(file), is_executable(exec) {};
+  const char* file_name;
+  bool is_executable;
+};
+
+static const std::map<std::string, SymbolAsset> g_symbol_asset_map = {
```

Comment:
This creates a static intializer and an exit-time destructor.  Instead, I'd just use a flat array and use `lower_bound` to query it.